feat: add webauthn signal api serializers

joostdebruijn opened the pull request:
Target branch: 5.3.x
Resolves issue: n/a

<!-- replace space with "x" in square brackets: [x] -->
- [ ] It is a Bug fix
- [x] It is a New feature
- [ ] Breaks BC
- [ ] Includes Deprecations

The WebAuthn Signal API is a new extension to WebAuthn that allows servers and clients to exchange additional “signal” data in a privacy-preserving way, enabling scenarios such as passkey upgrade flows, security key cross-device operations, seamless migration, and more. See the [W3C explainer](https://github.com/w3c/webauthn/wiki/Explainer:-WebAuthn-Signal-API-explainer) and [Chrome Developer Docs](https://developer.chrome.com/docs/identity/webauthn-signal-api).

This pull request introduces new denormalizers and corresponding signal classes to support WebAuthn serialization for the Signal API. These changes handle three signals: `AllAcceptedCredentials`, `CurrentUserDetails`, and `UnknownCredential`.

joostdebruijn commented:
I'm not sure what to do with the failing checks, because some checks are failing on files I didn't touch.

Spomky commented:
Hi,

Many thanks. I was not aware of these signals. I read the documentation carefully and it looks really interesting.

I will investigate the failing tests.

Spomky committed:
Add signal classes and denormalizers for WebAuthn events

Introduced `Signal` interface along with three implementations: `AllAcceptedCredentials`, `CurrentUserDetails`, and `UnknownCredential`. Added corresponding denormalizers to handle serialization and deserialization of these events, updating `WebauthnSerializerFactory` to register them. Extended `SerializerTest` to validate proper functionality with new test cases.

Spomky committed:
Update PHPStan baseline with new rules for Signal denormalizers

Expanded the PHPStan baseline to include new warnings related to type handling and offset accessibility in Signal denormalizers (`SignalAllAcceptedCredentialsDenormalizer`, `SignalCurrentUserDetailsDenormalizer`, and `SignalUnknownCredentialDenormalizer`), ensuring accurate tracking of static analysis issues.

Spomky commented:
Many thanks !
